### ISSUE TITLE
47.14 Player units not making PSRs for exceeding normal MP in low gravity?

### ISSUE THREAD

**Issue by gsparks3** [MegaMek/megamek#2431]
### Environment
This issue was first noticed in MekHQ integrated Megamek, but reproduces when loading the save in standalone Megamek.
47.14
Windows 10
Java 11.0.8

### Description
While playing an AtB campaign scenario on a planet with low gravity, I noticed that (unlike Princess's units) my own mechs did not seem to be making PSRs to avoid leg damage while moving at speeds above their normal 1G MP. Alternately, checks are being made but not displayed in the movement-phase round report. 

Reproducible in the provided savegame - replace the bot units, then run all of the player mechs forward at maximum speed (which exceeds their 1G walk MP). Observe that no PSRs are listed in the movement-phase round report, despite MM warning that PSRs will be required for such movement. 

### Files
[testLowGMovement.sav.gz](https://github.com/MegaMek/megamek/files/5594439/testLowGMovement.sav.gz)
Player name: Greyscales
No errors in log files.

**Comment by gsparks3** [comment]
Update: In a different fight, my units made PSRs for _jumping_ further than they could in 1G, but not _running_ further than they could in 1G.

**Comment by neoancient** [comment]
I worked through this far enough to tell that the PSRs are being made but not showing up in the round report. When I started looking into why they're not showing in the round report they started showing up and now I can't reproduce the issue.

**Comment by gsparks3** [comment]
New contract was on a planet with even lower gravity, so I thought I'd toss another save your way, except on checking the PSRs are showing up for me too, now. What is this wizardry? I've seen stuff start working when the technician looks at it, but normally it doesn't work remotely ...

I'll keep an eye out for if the issue comes back.

**Comment by gsparks3** [comment]
Update: That was fast. On this new planet, my units can _walk_ faster than they normally run, and that seems to trigger the issue. I'm not sure why it suddenly started working if they run, however. Save attached. 

Player name: Greyscales
[testLowGMovement2.sav.gz](https://github.com/MegaMek/megamek/files/5608566/testLowGMovement2.sav.gz)

**Comment by neoancient** [comment]
I'm sure it's a quantum effect. The act of observation changes the outcome. In your first save I was able to trigger the issue reliably by using full sprinting movement, until I started running under a debugger and it worked like intended.

**Comment by neoancient** [comment]
Based on some more rigid testing I did today, making sure I kept doing the same thing, I think it comes down to two things: the client and the server have different understandings of the gravity rules, and how the gravity and sprinting rules interact. On the client side, if you get a popup nag if you move faster than you normally could for the mode (walk/run/sprint). On the server side, you roll a PSR if you move faster than your maximum move.

So for instance, for a unit that normally moves 4/6, but moves 5/8 under reduced gravity:
Client side: walking 1-5, running 6-8, sprinting 9-10. PSR at 5 (walking > 4), 7-8 (running > 6), or 9-10 (sprinting > 8).
Server side: PSR at 9-10, which is faster than the normal top (sprinting) speed of 8.

I believe the client is wrong. TacOps says the PSR is needed if it exceeds the running or flanking speed at 1G, so there is no such thing as a PSR for walking too fast. The question is how the sprinting and gravity rules work together. The way the server works now, the PSR is only required if the unit exceeds the 1G sprinting speed. With a strict reading of the TacOps rule, running 7-8 or sprinting 9-10 would require the PSR.
